libs/pageserver_api/src/models.rs

```diff
@@ -60,6 +62,7 @@ impl TenantState {
             // tenant mgr startup distinguishes attaching from loading via marker file.
             // If it's loading, there is no attach marker file, i.e., attach had finished in the past.
             Self::Loading => Attached,
+            Self::Activating => todo!(),
```

Comment:
if `Loading` is `Attached` I can't see any other way for `Activating` than to be `Attached`

Reply:
Hm, true, gotta think of this in light of https://neondb.slack.com/archives/C033A2WE6BZ/p1684866965026069?thread_ts=1684854161.739799&cid=C033A2WE6BZ as well

Reply:
So right now "not found from s3" tenant on attach will yield broken and is assumed to *stay* broken, as in we will not get rid of the tenant in memory? Confirmed we don't.

For me the answer seems to be quite simple; just forget everything about the tenant if we never had any possibility of attaching it? Problem will probably be then crash-safety, because we will have the files on disk. It seems even the attach marker was left, so until a manual detach comes we can't make progress even across restarts. This is most likely current design.

I am unsure if we *need* to focus on solving the "tried to attach tenant from different region" issue.

Reply:
I'll put this in the right issue

Reply:
https://github.com/neondatabase/neon/issues/4344#issuecomment-1563947761

Reply:
Unresolved because this is already an issue on this PR: https://github.com/neondatabase/neon/pull/4321#issuecomment-1566729898

Reply:
IMHO we could just set this to `Attached` right away, because we do require the manual detach before this. If we ever slim down the amount of reasons for `Attached`, it would be fine as well.

Reply:
Solved as "attachment_status is same as the state where we arrived to Activating" in [d6d0c74](https://github.com/neondatabase/neon/pull/4321/commits/d6d0c74c0699a3ff0c7c977a4c46c29754c589f0)

Reply:
Green light from me

Reply:
Thanks. CI is taking forever again. Resolving this.